src/main/java/org/zalando/nakadi/validation/JsonSchemaLoader.java

```diff
@@ -0,0 +1,191 @@
+package org.zalando.nakadi.validation;
+
+import com.google.common.collect.Lists;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.zalando.nakadi.domain.CompatibilityMode;
+import org.zalando.nakadi.domain.EventType;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class JsonSchemaLoader {
+    public static final String DATA_CHANGE_WRAP_FIELD = "data";
+
+    private static final String ADDITIONAL_PROPERTIES = "additionalProperties";
+    private static final String ADDITIONAL_ITEMS = "additionalItems";
+    private final List<String> keywordsWithNestedSchemas = Lists.newArrayList("definitions", "dependencies",
```

Comment:
Is this list changeable? I suppose not, maybe we can make it static. also makes sense to use set vs list.